nextjs-template/src/services/notehub/NotehubDataProvider.ts

```diff
@@ -36,6 +33,24 @@ export const getBestLocation = (object: NotehubLocationAlternatives) =>
     throw new Error("Method not implemented.");
   }
 
+  async getFleetsByProject(): Promise<any> {
+    const fleetsByProject = await this.notehubAccessor.getFleetsByProject();
+    return fleetsByProject;
+  }
+
+  async getDevicesByFleet(fleetUID: string): Promise<any> {
+    const devicesByFleet = await this.notehubAccessor.getDevicesByFleet(
+      fleetUID
+    );
+    return devicesByFleet;
+  }
+
+  async getFleetEnvVars(fleetUID: string): Promise<any> {
+    const fleetEnvVars = await this.notehubAccessor.getFleetEnvVars(fleetUID);
+    // attach fleet UID to env vars for combining data later
+    return { fleetUID: fleetUID[0], ...fleetEnvVars };
```

Comment:
I'm not sure what the array index on fleetUID does. Possibly a hangover from earlier code?

Reply:
You're correct - that was definitely some bad code from me not properly looping through an array of promises in the main `index.ts` page. I've fixed it now and all the subsequent code affected by it

Reply:
No worries, I know it's WiP. :)